src/game/Threat/ThreatManager.cpp

```diff
@@ -316,17 +316,22 @@ HostileReference* ThreatContainer::selectNextVictim(Creature* pAttacker, Hostile
                 continue;
             }
 
-            float attackDistance = pAttacker->GetMaxChaseDistance(target);
             // Skip this unit if low priority
-            if (!allowLowPriorityTargets && (target->IsImmuneToDamage(pAttacker->GetMeleeDamageSchoolMask()) ||
-                                            target->IsSecondaryThreatTarget() ||
-                                           (attackerImmobilized && !target->IsWithinDist(pAttacker, attackDistance))))
+            if (!allowLowPriorityTargets)
             {
-                // current victim is a second choice target, so don't compare threat with it below
-                if (currentRef == pCurrentVictim)
-                    pCurrentVictim = nullptr;
-                ++iter;
-                continue;
+                float attackDistance = pAttacker->GetMaxChaseDistance(target);
+                // second choice targets are: immune to attacker's autoattack damage school / is secondary threat target (fear, gouge etc) /
+                // is outside of attacker's caster chase distance if rooted caster / is unreachable by attacker's melee attacks if rooted melee.
+                if ( target->IsImmuneToDamage(pAttacker->GetMeleeDamageSchoolMask()) || (target->IsSecondaryThreatTarget()) ||
```

Comment:
why is (target->IsSecondaryThreatTarget()) wrapped in brackets, its just 1 function call, doesn't need brackets